plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui.test/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/internal/AppEngineApplicationQueryJobTest.java

```diff
@@ -102,6 +103,24 @@ public void testRun_projectHasNoApplication()
 
     assertEquals(AppEngine.NO_APPENGINE_APPLICATION, project.getAppEngine());
   }
+  
+
+  @Test
+  public void testRun_accountDoesNotHavePermission()
+      throws ProjectRepositoryException, InterruptedException {
+    when(projectRepository.getAppEngineApplication(credential, "projectId"))
+        .thenThrow(new ApplicationPermissionsException("testException", null));
+
+    queryJob.schedule();
+    queryJob.join();
+
+    verify(isLatestQueryJob).apply(queryJob);
+    verify(projectSelector).isDisposed();
+    verify(projectSelector).setStatusLink(
+        "This account does not have permission for the App Engine application in project projectId.", null);
```

Comment:
`for` → `to access`?  `for` seems nebulous, but perhaps that's a good thing?

Reply:
Maybe "to deploy to"?
The account may be able to access the app, just not replace it.

Reply:
I prefer "for" then since we're not actually trying to deploy at this particular case.  It's probably the case that if I can't retrieve the app details, I can't deploy an app either?

Reply:
Just for clarity, an "application" here means the [Application resource](https://cloud.google.com/appengine/docs/admin-api/creating-an-application) that may exist at most one per GCP project. So the query here is to check if the project has ever created the application, to start using App Engine.